pkg/kubelet/config/common.go

```diff
@@ -73,6 +73,20 @@ func applyDefaults(pod *api.Pod, source string, isFile bool, nodeName types.Node
 	// The generated UID is the hash of the file.
 	pod.Annotations[kubetypes.ConfigHashAnnotationKey] = string(pod.UID)
 
+	if isFile {
```

Comment:
This is a cautious decision not introducing this to http source. Honestly, I don't know the use case of http-based pods, and it is arguable that they have their own controller outside the node.